tidy3d/components/time_modulation.py

```diff
@@ -0,0 +1,207 @@
+""" Defines time modulation to the medium"""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Union
+from math import isclose
+
+import pydantic.v1 as pd
+import numpy as np
+
+from .base import Tidy3dBaseModel, cached_property
+from .types import InterpMethod
+from .source import AbstractTimeDependence
+from .data.data_array import SpatialDataArray
+from ..exceptions import ValidationError
+from ..constants import HERTZ
+
+
+class AbstractTimeModulation(AbstractTimeDependence):
```

Comment:
inherit this from `ABC` (and other abstract ones, eg. AbstractTimeDependence, if not already)